### DISCUSSION TITLE
2.3.90 20211213 so-allow lxml missing

### DISCUSSION THREAD

**Discussion by darth-drew** [Security-Onion-Solutions/securityonion#6598] [General] [answered]
It seems that the so-allow script underwent bash to python overhaul in the upgrade from 2.3.80 to 2.3.90 20211213. When trying to run the script now on a distributed airgapped install, I am receiving "ModuleNotFoundError: no module named 'lxml'". We were able to install lxml using our local repository and run the script, but I suspect this was meant to be included in the ISO.

**Comment by TOoSmOotH** [comment]
When you say "our local repository" are you not using the auto updated one on the manager?

**Comment by darth-drew** [comment]
The auto-updated one is definitely present but I did not try to pull from it directly. After the update, I assumed the lxml packages would have been installed automatically, allowing us to run the so-allow script. Is this a bad assumption? "Our local repository" was referring to one external to SO.

**Comment by dougburks** [comment, marked as answer]
I just tested a 2.3.80 airgap installation and can confirm that it upgraded successfully using the securityonion-2.3.90-20211213 ISO image. The `python36-lxml` package was automatically installed and `so-allow` works properly.

You might try a test installation of 2.3.80 in your environment and then updating it to 2.3.90 to then compare to your existing deployment.
